src/API/Functions/UnitMembers.cs

```diff
@@ -0,0 +1,115 @@
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using System.Threading.Tasks;
+using CSharpFunctionalExtensions;
+
+using API.Data;
+using API.Middleware;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Azure.WebJobs.Extensions.OpenApi.Core.Attributes;
+using System.Net;
+using System.Collections.Generic;
+using Models;
+using Microsoft.OpenApi.Models;
+using System.Net.Mime;
+using System.Linq;
+
+namespace API.Functions
+{
+	public static class UnitMembers
+	{
+		public const string UnitMembersTitle = "Unit Memberships";
+
+
+		[FunctionName(nameof(UnitMembers.UnitMembersGetAll))]
+		[OpenApiOperation(nameof(UnitMembers.UnitMembersGetAll), UnitMembersTitle, Summary = "List all unit memberships")]
+		[OpenApiResponseWithBody(HttpStatusCode.OK, MediaTypeNames.Application.Json, typeof(List<UnitMemberResponse>), Description = "A collection of unit membership record")]
+		public static Task<IActionResult> UnitMembersGetAll(
+			[HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = "memberships")] HttpRequest req)
+			=> Security.Authenticate(req)
+				.Bind(_ => UnitMembersRepository.GetAll())
+				.Bind(res => Pipeline.Success(res.Select(e => e.ToUnitMemberResponse())))
+				.Finally(dtos => Response.Ok(req, dtos));
+
+		[FunctionName(nameof(UnitMembers.UnitMembersGetOne))]
+		[OpenApiOperation(nameof(UnitMembers.UnitMembersGetOne), UnitMembersTitle, Summary = "Find a unit membership by ID")]
+		[OpenApiParameter("membershipId", Type = typeof(int), In = ParameterLocation.Path, Required = true, Description = "The ID of the unit membership record.")]
+		[OpenApiResponseWithBody(HttpStatusCode.OK, "application/json", typeof(UnitMemberResponse), Description = "A unit membership record")]
+		[OpenApiResponseWithBody(HttpStatusCode.NotFound, MediaTypeNames.Application.Json, typeof(ApiError), Description = "No unit membership was found with the ID provided.")]
+		public static Task<IActionResult> UnitMembersGetOne(
+			[HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = "memberships/{membershipId}")] HttpRequest req, int membershipId)
+			=> Security.Authenticate(req)
+				.Bind(_ => UnitMembersRepository.GetOne(membershipId))
+				.Bind(res => Pipeline.Success(res.ToUnitMemberResponse()))
+				.Finally(result => Response.Ok(req, result));
+
+		[FunctionName(nameof(UnitMembers.CreateUnitMembers))]
+		[OpenApiOperation(nameof(UnitMembers.CreateUnitMembers), UnitMembersTitle, Summary = "Create a unit membership", Description = "Authorization: Unit memberships can be created by any unit member that has either the `Owner` or `ManageMembers` permission on their unit membership. See also: [Units - List all unit members](#operation/UnitsGetAll).")]
+		[OpenApiRequestBody(MediaTypeNames.Application.Json, typeof(UnitMemberRequest), Required = true)]
```

Comment:
The [generated documentation](http://localhost:8080/#operation/CreateUnitMembers) doesn't show `Role` as required, and shows the default value for `permissions` as "owner" when it's actually "viewer".

I'm not sure if it's worth tying ourselves in knots to get OpenApi honoring these data annotations, but I just wanted to point it out.

Reply:
As discussed, all the documentation presently is like this. Requireds only show for path parameters like relationshipId. The default values just seem to be ignored as well. I'm not sure, but I thought the `/// some description` where also supposed to map but don't seem to work either.